modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -495,11 +495,17 @@ def validate_form_526_service_information!(target_veteran)
       end
 
       def validate_claim_date_to_active_duty_end_date!(service_information)
+        ant_sep_date = form_attributes['serviceInformation']['federalActivation']['anticipatedSeparationDate']
         max_period = service_information['servicePeriods'].max_by { |sp| sp['activeDutyEndDate'] }
-        date = form_attributes['claimDate'] || Time.find_zone!('Central Time (US & Canada)').today
-        if Date.strptime(date.to_s, '%Y-%m-%d') > Date.strptime(max_period['activeDutyEndDate'], '%Y-%m-%d') + 180.days
+        date = form_attributes['claimDate']
+        if (Date.strptime(max_period['activeDutyEndDate'],
+                          '%Y-%m-%d') + 180.days < Date.strptime(date.to_s, '%Y-%m-%d') &&
+                          max_period['activeDutyEndDate'].present?) ||
+           (Date.strptime(ant_sep_date,
+                          '%Y-%m-%d') + 180.days < Date.strptime(date.to_s, '%Y-%m-%d') &&
+                          ant_sep_date.present?)
```

Comment:
@stiehlrod since they are not required do we need a check to see if `activeDutyEndDate ` or `anticipatedSeparationDate ` are present before calling `strptime` ?

Reply:
Added a check for 503-506, and created a ticket to create a check for the confinement dates section, [here](https://jira.devops.va.gov/browse/API-31580)

Reply:
If I submit without an `activeDutyEndDate` value the `max_period` appears to be set at ````
{"serviceBranch"=>"Public Health Service", "activeDutyBeginDate"=>"2008-11-14", "serviceComponent"=>"Active", "separationLocationCode"=>"25077"}
````

So I still get `no implicit conversion of nil into String"` from line 511